deeprank2/molstruct/residue.py

```diff
@@ -32,58 +35,51 @@ def __init__(
         self._atoms = []
 
     def __eq__(self, other) -> bool:
-        return (
-            isinstance(self, type(other))
-            and self._chain == other._chain
-            and self._number == other._number
-            and self._insertion_code == other._insertion_code
-        )
+        if isinstance(other, Residue):
+            return (self._chain == other._chain
+                    and self._number == other._number
+                    and self._insertion_code == other._insertion_code
+            )
+        return NotImplemented
 
     def __hash__(self) -> hash:
         return hash((self._number, self._insertion_code))
 
     def get_pssm(self) -> PssmRow:
-        """
-        If the residue's chain has pssm info linked to it,
-        then return the part that belongs to this residue.
-        """
-
+        """Load pssm info linked to the residue."""
         pssm = self._chain.pssm
         if pssm is None:
             raise FileNotFoundError(f'No pssm file found for Chain {self._chain}.')
-
         return pssm[self]
 
     @property
     def number(self) -> int:
         return self._number
 
     @property
-    def chain(self):
+    def chain(self) -> Chain:
         return self._chain
 
     @property
     def amino_acid(self) -> AminoAcid:
         return self._amino_acid
 
     @property
-    def atoms(self):
+    def atoms(self) -> list["Atom"]:
         return self._atoms
 
     @property
     def number_string(self) -> str:
-        "contains both the number and the insertion code (if any)"
-
+        """Contains both the number and the insertion code (if any)."""
         if self._insertion_code is not None:
             return f"{self._number}{self._insertion_code}"
-
         return str(self._number)
 
     @property
     def insertion_code(self) -> str:
         return self._insertion_code
 
-    def add_atom(self, atom):
+    def add_atom(self, atom: "Atom"):
```

Comment:
Okay I guess the quotation marks are because of the TYPE_CHECKING thing, but I'll leave this comment here as well just in case